lib/src/commonMain/kotlin/cash/z/ecc/android/bip39/Mnemonics.kt

```diff
@@ -361,13 +359,9 @@ fun WordCount.toEntropy(): ByteArray = ByteArray(bitLength / 8).apply {
 // Private Extensions
 //
 
-private fun ByteArray?.toSha256() = MessageDigest.getInstance("SHA-256").digest(this)
+private fun ByteArray.toSha256() : ByteArray = this.toByteString().sha256().toByteArray()
```

Comment:
Okio already does that according to their docs. https://github.com/square/okio/blob/master/docs/multiplatform.md

```
[](https://github.com/square/okio/blob/master/docs/multiplatform.md#hashing)Hashing

Okio includes Kotlin implementations of MD5, SHA-1, SHA-256, and SHA-512. This includes both hash functions and HMAC functions.

Okio uses the built-in implementations of these functions on the JVM.
```

Reply:
Our goal is to have zero third party dependencies, beyond the Kotlin standard library and underlying platform the code is running on (e.g. JVM or Apple native frameworks).  We don't plan to ship a version of our library that depends on Okio.

So the comment here is really to enable stage 3 (multiplatform with only actual implementation for JVM), while keeping the path open for someone to create a fork leveraging Okio for other platforms as a stopgap.  Such an approach would minimize the difficulty of incorporating upstream changes, since it was written with such usage in mind.

Since you mentioned that Apple platforms were your priority, it looks like there might be an actual native implementation that can be provided for Apple platforms without needing Okio https://github.com/JetBrains/kotlin-native/issues/2466#issuecomment-447225892

Assuming that works without requiring any dependencies, we could partial multiplatform support (JVM and Apple platforms) then add other platforms over time (or again enable forks that decide to incorporate Okio).